File: RecordTechnology/RecordTechnology/201711/HXAppPod/Utils/HXFileHelper.m
//
//  HXFileHelper.m
//  store
//
//  Created by Arthur Wang on 14-10-14.
//  Copyright (c) 2014年 Arthur Wang All rights reserved.
//

#import "HXFileHelper.h"
#define HX_DATA_PATH @"HXData"

@implementation HXFileHelper

+ (NSString *)absolutePathForDataFile:(NSString *)fileName {
    NSArray *dirs = NSSearchPathForDirectoriesInDomains(NSLibraryDirectory, NSUserDomainMask, YES);
    if (dirs == nil || [dirs count] == 0)
        return nil;
    NSString *path = [[dirs objectAtIndex:0] stringByAppendingPathComponent:HX_DATA_PATH];
    
    if (![[NSFileManager defaultManager] fileExistsAtPath:path]) {
        NSError *error = nil;
        [[NSFileManager defaultManager] createDirectoryAtPath:path
                                  withIntermediateDirectories:YES
                                                   attributes:nil
                                                        error:&error];
        if (error != nil) {
//            HXLogError(@"HXFileHelper", @"Failed to create directory: %@", path);
            return nil;
        } else
            return [path stringByAppendingPathComponent:fileName];
    } else
        return [path stringByAppendingPathComponent:fileName];
}

+ (BOOL)existsDataFile:(NSString *)fileName {
    return [[NSFileManager defaultManager] fileExistsAtPath:[self absolutePathForDataFile:fileName]];
}

+ (void)writeDataFile:(NSString *)fileName withData:(NSData *)data {
    NSString *absFile = [self absolutePathForDataFile:fileName];
    [data writeToFile:absFile atomically:YES];
}

+ (NSString *)absolutePathForDocumentFile:(NSString *)docFile {
    NSArray *dirs = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
    if (dirs == nil || [dirs count] == 0)
        return nil;
    else
        return [[dirs objectAtIndex:0] stringByAppendingPathComponent:docFile];
}

+ (void)writeDocumentFile:(NSString *)fileName withData:(NSData *)data {
    NSString *absFile = [self absolutePathForDocumentFile:fileName];
    [data writeToFile:absFile atomically:YES];
}

+ (BOOL)existsDocumentFile:(NSString *)fileName {
    return [[NSFileManager defaultManager] fileExistsAtPath:[self absolutePathForDocumentFile:fileName]];
}

@end

File: RecordTechnology/RecordTechnology/201711/HXAppPod/Utils/HXFileHelper.swift
//
//  HXFileHelper.swift
//  RecordTechnology
//
//  Created by ArthurWang on 2017/11/24.
//  Copyright © 2017年 Monkey King. All rights reserved.
//

import Foundation

class HXFileHelper: NSObject {

    static let HX_DATA_PATH:String = "HXData";

    @objc class func absolutePathForDataFile(fileName: String) -> String? {

        let dirs = NSSearchPathForDirectoriesInDomains(FileManager.SearchPathDirectory.libraryDirectory, FileManager.SearchPathDomainMask.userDomainMask, true);
        if 0 == dirs.count {
            return nil;
        }

        let path = (dirs[0] as NSString).appendingPathComponent(HX_DATA_PATH);
        if (FileManager.default.fileExists(atPath: path)) {
            return (path as NSString).appendingPathComponent(fileName);
        } else {
            do {
                try FileManager.default.createDirectory(atPath: path, withIntermediateDirectories: true, attributes: nil);

                return (path as NSString).appendingPathComponent(fileName);
            } catch {
                return nil;
            }
        }
    }

    @objc class func existsDataFile(fileName: String) -> Bool {
        let path = self.absolutePathForDataFile(fileName: fileName);

        if nil == path {
            return false;
        }

        return FileManager.default.fileExists(atPath: path!);
    }

    @objc class func writeDataFile(fileName: String, data: NSData) {
        let absFile = self.absolutePathForDataFile(fileName: fileName);

        if nil == absFile {
            return;
        }

        data.write(toFile: absFile!, atomically: true);
    }

    @objc class func absolutePathForDocumentFile(docFile: String) -> String? {
        let dirs = NSSearchPathForDirectoriesInDomains(FileManager.SearchPathDirectory.documentDirectory, FileManager.SearchPathDomainMask.userDomainMask, true);
        if 0 == dirs.count {
            return nil;
        }

        return (dirs[0] as NSString).appendingPathComponent(docFile);
    }

    @objc class func writeDocumentFile(fileName: String, data: NSData) {
        let absFile = self.absolutePathForDocumentFile(docFile: fileName);

        if nil == absFile {
            return;
        }

        data.write(toFile: absFile!, atomically: true);
    }

    @objc class func existsDocumentFile(fileName: String) -> Bool {
        let path = self.absolutePathForDocumentFile(docFile: fileName);

        if nil == path {
            return false;
        }

        return FileManager.default.fileExists(atPath:path!);
    }



}
